src/notebooklm/_source_add.py

```
@@ -193,15 +210,63 @@ async def add_drive(
+        # false-positive substring matches against other file_ids that
+        # happen to contain ``<file_id>`` as an interior substring.
+        drive_url_marker = f"/d/{file_id}"
+
+        async def _probe() -> Source | None:
+            try:
+                sources = await list_sources(notebook_id)
+            except (AuthError, RateLimitError, ServerError, NetworkError):
+                # Transport- and auth-level probe failures must propagate
+                # — see the rationale in ``add_url._probe`` (P1-2).
+                raise
+            except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)